CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportTopComponent.java

```diff
@@ -0,0 +1,894 @@
+                                break;
+                default:    break;                                    
+            }
+        }
+    }
+
+    public void updateSessionErrorsBox(final int insertPos) {
+        // get current snapshot of errors
+        boolean sessionUpdated = false;
+        while (!sessionUpdated) {
+            if (updateInProgress.compareAndSet(false, true)) {
+                refreshSessionErrors();
+                final Date nextFilterDate = new Date();
+                final int errCount = sessionErrors.size();
+                // rebuild                
+                sessionErrorsBox.getChildren().clear();
+                for (int i = 0; i < errCount; i++) {
+                    boolean allowPopupDisplay = false;
+                    if (errorReportRunning && (("SEVERE".equals(sessionErrors.get(i).getErrorLevel().getName()) && severePopCheckBox.isSelected())
+                            || ("WARNING".equals(sessionErrors.get(i).getErrorLevel().getName()) && warningPopCheckBox.isSelected())
+                            || ("INFO".equals(sessionErrors.get(i).getErrorLevel().getName()) && infoPopCheckBox.isSelected())
+                            || ("FINE".equals(sessionErrors.get(i).getErrorLevel().getName()) && finePopCheckBox.isSelected()))) {
+                        allowPopupDisplay = true;
+                    }
+                    sessionErrorsBox.getChildren().add(generateErrorReportTitledPane(sessionErrors.get(i)));
+                    if (allowPopupDisplay && sessionErrors.get(i).getLastDate().after(filterUpdateDate)) {
+                        ErrorReportDialogManager.getInstance().showErrorDialog(sessionErrors.get(i));
+                    }
+                }
+                
+                // check if popup needed on hidden entries ... these entries can only exist with the allow popup checkbox ticked
+                if (errorReportRunning) {
+                    for (final ErrorReportEntry entry : hiddenErrors) {
+                        if (entry.getLastDate().after(filterUpdateDate)) {
+                            ErrorReportDialogManager.getInstance().showErrorDialog(entry);
+                        }
+                    }
+                }
+                final Date resumptionDate = ErrorReportDialogManager.getInstance().getGracePeriodResumptionDate();
+                if (resumptionDate == null || nextFilterDate.after(resumptionDate)) {
+                    filterUpdateDate = new Date(nextFilterDate.getTime());
+                } else {
+                    waitForGracePeriod = true;
+                }
+                sessionUpdated = true;
+                updateInProgress.set(false);
+            } else {
+                try {
+                    // session is being updated on another thread
+                    Thread.sleep(360);
+                } catch (final InterruptedException ex) {
+                    Exceptions.printStackTrace(ex);
+                }
```

Comment:
this may be impacted by the above but can we add a thread interrupt for this exception and replace the stace trace print with a log entry (include exception as parameter for entry)